atomic_reactor/plugins/pre_check_and_set_platforms.py

```diff
@@ -20,21 +20,41 @@
 from atomic_reactor.constants import PLUGIN_CHECK_AND_SET_PLATFORMS_KEY
 
 
+class MustBuildForAmd64(Exception):
+    """ Platforms must include one for GOARCH amd64 """
+
+
 class CheckAndSetPlatformsPlugin(PreBuildPlugin):
     key = PLUGIN_CHECK_AND_SET_PLATFORMS_KEY
     is_allowed_to_fail = False
 
-    def __init__(self, tasker, workflow, koji_target):
+    def __init__(self, tasker, workflow, koji_target, goarch):
 
         """
         constructor
 
         :param tasker: DockerTasker instance
         :param workflow: DockerBuildWorkflow instance
+        :param koji_target: str, Koji build target name
+        :param goarch: Mapping, platform name to GOARCH platform name
         """
         # call parent constructor
         super(CheckAndSetPlatformsPlugin, self).__init__(tasker, workflow)
         self.koji_target = koji_target
+        self.goarch = goarch
```

Comment:
This is available in reactor config.
Since this plugin is new to arrangement 6 which requires reactor config map, I don't think it's worth adding a fallback option to plugin parameter.

Reply:
Done.